### ISSUE TITLE
mailwatch_postfix_relay script throwing errors

### ISSUE THREAD

**Issue by gordoncolyn** [mailwatch/MailWatch#924]
/usr/bin/php /usr/local/bin/mailwatch_postfix_relay.php --refresh

When I run the above script I get the following errors;

```php
Fatal error: Uncaught exception 'mysqli_sql_exception' with message 'Incorrect time value: '68' for column 'delay' at row 1' in /var/www/html/mailscanner/functions.php:968
Stack trace:
#0 /var/www/html/mailscanner/functions.php(968): mysqli->query('REPLACE INTO mt...')
#1 /var/www/html/mailscanner/mtalogprocessor.inc.php(143): dbquery('REPLACE INTO mt...')
#2 /usr/local/bin/mailwatch_postfix_relay.php(94): MtaLogProcessor->doit('cat /var/log/ma...')
#3 {main}
  thrown in /var/www/html/mailscanner/functions.php on line 968
cat: write error: Broken pipe
```

Server
Centos 6.9
Apache 2.2.15
Mailwatch 1.2.5 - installed from zip file upgrading from 1.2.3, and then 1.2.4
Mysql Server 5.7.19
Postfix 2.6.6

How can I sort out the problem?

**Comment by Skywalker-11** [comment]
Do you also get that error if you run it without the `--refresh` and then receive a mail?

**Comment by gordoncolyn** [comment]
No, same error;
/usr/bin/php /usr/local/bin/mailwatch_postfix_relay.php

Fatal error: Uncaught exception 'mysqli_sql_exception' with message 'Incorrect time value: '67' for column 'delay' at row 1' in /var/www/html/mailscanner/functions.php:968
Stack trace:
#0 /var/www/html/mailscanner/functions.php(968): mysqli->query('REPLACE INTO mt...')
#1 /var/www/html/mailscanner/mtalogprocessor.inc.php(143): dbquery('REPLACE INTO mt...')
#2 /usr/local/bin/mailwatch_postfix_relay.php(97): MtaLogProcessor->doit('cat /var/log/ma...')
#3 {main}
  thrown in /var/www/html/mailscanner/functions.php on line 968
cat: write error: Broken pipe

**Comment by gordoncolyn** [comment]
Not sure if this may be related but I had this error when running php upgrade.php;
 - Fix schema for timestamp field in `maillog` table.................. ERROR
Database error: Invalid default value for 'timestamp' - SQL = 'ALTER TABLE `maillog` CHANGE `timestamp` `timestamp` TIMESTAMP NOT NULL DEFAULT 0'

So i commented out these 2 lines to be able to complete the upgrade...
    //$sql1 = 'ALTER TABLE `maillog` CHANGE `timestamp` `timestamp` TIMESTAMP NOT NULL DEFAULT 0';
    //$sql2 = 'ALTER TABLE `maillog` ALTER COLUMN `timestamp` DROP DEFAULT';

**Comment by Skywalker-11** [comment]
Its not related. It is table mtalog not maillog thats affected here

@gordoncolyn can paste the mail log entries for a message? Especially the line that contains `delay` or look like this `Aug 30 22:43:51 master postfix/pipe[3260]: XXXXXXXX: to=<xxxx>, relay=XXX, delay=103, delays=102/0.15/0/0.62, dsn=2.0.0, status=sent`

**Comment by gordoncolyn** [comment]
ok, here is an example;
Aug 27 09:11:59 xxxx postfix/smtp[12434]: xxxx: to=<xxxx>, relay=xxxx[xxxx]:25, delay=35, delays=28/0/5/1.6, dsn=2.0.0, status=sent (250 2.0.0 xxxx Message accepted for delivery)

**Comment by asuweb** [comment]
This is fixed in PR #942
